bloomstack_core/services/payments.py

```diff
@@ -44,6 +48,14 @@ def collect(amount, delivery_note, sales_invoice=None, returned_items=None):
 		if invoices:
 			sales_invoice = invoices[0].against_sales_invoice
 
+	if not sales_invoice: # check for reverse linking
```

Comment:
I see that you're checking if a sales invoice has a delivery note attached to it, but how exactly are you checking for reverse linking here?

Reply:
Delivery Note is mandatory parameter for this end point. here we are looking for entry in Sales Invoice Item which has same Delivery Note as given Parameter, If entry exists then we are fetching its Parent(The Sales Invoice itself)